nx/lib/nx/lin_alg.ex

```diff
@@ -358,7 +364,8 @@ defmodule Nx.LinAlg do
     # The idea is that by dividing the tensor by it, large values of
     # tensor entries and large values of p are reduced, which in turn
     # avoids numerical overflow.
-    numerical_stability_coefficient = Nx.reduce_max(abs_t)
+    # The second max prevents from division by zero.
+    numerical_stability_coefficient = Nx.reduce_max(abs_t) |> Nx.max(1.0e-16)
```

Comment:
Maybe a select is better?

```suggestion
    numerical_stability_coefficient = Nx.reduce_max(abs_t)
    numerical_stability_coefficient = Nx.select(Nx.greater_than(numerical_stability_coefficient, 0), numerical_stability_coefficient, 1)
```

Reply:
Yes, I changed the code to a version with separate defnp, but this version is actually the best.